### PR TITLE
Add try builder config for cocoon device doctor

### PR DESCRIPTION
**Pull request by keyonghan**
Cocoon device doctor recipe to build/upload device_doctor package to cipd is ready. This PR adds pre-submit builder configs.

These builders should be triggered whenever any updates from `cocoon/device_doctor`.

### CODE REVIEW

File: config/cocoon_config.star [hunk 1]
```diff
@@ -29,15 +30,29 @@ def cocoon_define_recipes():
...
     )
 
+def device_doctor_recipes():
+    # Defines recipes
+    luci.recipe(
+        name = "cocoon/device_doctor",
+        cipd_package = "flutter/recipe_bundles/flutter.googlesource.com/recipes",
+        cipd_version = "refs/heads/master",
+    )
+
```
Comment (godofredoc) [inline config/cocoon_config.star:40]: Can we consolidate the two methods in a generic one? eg.
recipe_names = [ "cocoon", "cocoon/device_doctor"] 
for recipe in recipe_names:
   luci.recipe(
         name = recipe ...

File: config/cocoon_config.star [hunk 2]
```diff
@@ -29,15 +30,29 @@ def cocoon_define_recipes():
         cipd_version = "refs/heads/master",
     )
 
+def device_doctor_recipes():
```
Comment (godofredoc) [inline config/cocoon_config.star:33]: can we also move the cocoon recipe to cocoon/cocoon?

File: config/cocoon_config.star [hunk 3]
```diff
@@ -46,5 +61,30 @@ def cocoon_try_config(platform_args):
         add_cq = True,
         **platform_args["linux"]
     )
+    common.linux_try_builder(
+        name = "Linux device_doctor|device_doctor",
+        recipe = "cocoon/device_doctor",
+        list_view_name = list_view_name,
+        repo = repos.COCOON,
+        add_cq = True,
```
Comment (godofredoc) [inline config/cocoon_config.star:69]: do we need the three builders blocking recipes changes? if not we can remove add_cq from some of them.

File: config/cocoon_config.star [hunk 4]
```diff
@@ -46,5 +61,30 @@ def cocoon_try_config(platform_args):
...
+    )
+
+    # Defines cocoon mac try builders
+    common.mac_try_builder(
```
Comment (godofredoc) [inline config/cocoon_config.star:73]: Do the test require phones? if yes we'll need to add additional dimensions.

File: config/cocoon_config.star (hunk 2 above)
Comment (keyonghan) [reply config/cocoon_config.star]: First recipe change: https://flutter-review.googlesource.com/c/recipes/+/9520
Also updated here to point to cocoon/cocoon.
Will land this PR when the above CL lands.

File: config/cocoon_config.star (hunk 1 above)
Comment (keyonghan) [reply config/cocoon_config.star:33]: Done.

File: config/cocoon_config.star (hunk 3 above)
Comment (keyonghan) [reply config/cocoon_config.star:62]: This is intended to make sure these three builders succeed when any changes from recipe side.

File: config/cocoon_config.star (hunk 4 above)
Comment (keyonghan) [reply config/cocoon_config.star:66]: Not now until we have capacity in try pool.

File: config/cocoon_config.star (hunk 3 above)
Comment (godofredoc) [reply config/cocoon_config.star:62]: I think the question is mac/linux use the same code path do we need to run both in led recipe tests?

**Review by godofredoc — APPROVED** [review]

File: config/cocoon_config.star (hunk 3 above)
Comment (keyonghan) [reply config/cocoon_config.star:62]: I think so. Though they are using the same code path, but they fetch different versions of dart, and then proceed to build.

**Review by CaseyHillers — APPROVED** [review]

**Pull request merged**